src/main/kotlin/sk/streetofcode/webapi/service/PostCommentServiceImpl.kt

```diff
@@ -0,0 +1,107 @@
+    val env: Environment
+) : PostCommentService {
+
+    companion object {
+        private val log = LoggerFactory.getLogger(PostCommentServiceImpl::class.java)
+    }
+
+    override fun getAll(postId: String): List<PostCommentDto> {
+        return postCommentRepository.findAllByPostId(postId).map { it.toPostCommentDto() }
+    }
+
+    override fun add(userId: String?, postId: String, addRequest: PostCommentAddRequest): PostCommentDto {
+        try {
+            val postComment = postCommentRepository
+                .save(
+                    PostComment(
+                        if (userId != null) socUserService.get(authenticationService.getUserId()) else null,
+                        postId,
+                        addRequest.postTitle,
+                        addRequest.commentText
+                    )
+                )
+
+            if (env.activeProfiles.contains("prod")) {
+                emailService.sendNewPostCommentNotification(postComment)
+            }
+
+            return postComment.toPostCommentDto()
+        } catch (e: Exception) {
+            log.error("Problem with saving postComment to db", e)
+            throw InternalErrorException("Could not save post comment")
+        }
+    }
+
+    override fun edit(
+        userId: String?,
+        postId: String,
+        commentId: Long,
+        editRequest: PostCommentEditRequest
+    ): PostCommentDto {
+        val comment = postCommentRepository.findById(commentId)
+            .orElseThrow { ResourceNotFoundException("Post comment with id $commentId was not found") }
+        if (postId != comment.postId) {
+            throw BadRequestException("Comment has different postId")
+        }
+
+        validateUserAuthorization(comment)
+
+        try {
+            comment.commentText = editRequest.commentText
+            comment.updatedAt = OffsetDateTime.now()
+
+            return postCommentRepository.save(comment).toPostCommentDto()
+        } catch (e: Exception) {
+            log.error("Problem with editing lectureComment", e)
+            throw InternalErrorException("Could not edit lecture comment")
+        }
+    }
+
+    override fun delete(userId: String?, postId: String, commentId: Long) {
+        val comment = postCommentRepository.findById(commentId)
+            .orElseThrow { ResourceNotFoundException("Post comment with id $commentId was not found") }
+        if (postId != comment.postId) {
+            throw BadRequestException("Comment has different postId")
+        }
+
+        validateUserAuthorization(comment)
+
+        postCommentRepository.deleteById(commentId)
+    }
+
+    // Only user who written the comment and admin can manipulate with it
+    private fun validateUserAuthorization(comment: PostComment) {
+        if (comment.socUser != null) {
+            val userId = authenticationService.getUserId()
+            if (comment.socUser.firebaseId != userId && !authenticationService.isAdmin()) {
```

Comment:
Shouldn't perhaps Admins be able to delete any comment? I think this would also impact FE, but to be honest, I think I can live with deleting them in the DB directly if we ever need to delete a comment.

Reply:
I believe admin is able to delete any comment. At least that's how I understand the if statement. But support from FE is not there. And yeah, deleting in the DB is okay if we will ever need it imho.

Reply:
Sorry, my bad, I completely missed the `isAdmin` check 🤦